Lib/test/test_urlparse.py

```diff
@@ -653,7 +653,7 @@ def test_attributes_bad_port(self):
         """Check handling of invalid ports."""
         for bytes in (False, True):
             for parse in (urllib.parse.urlsplit, urllib.parse.urlparse):
-                for port in ("foo", "1.5", "-1", "0x10"):
+                for port in ("foo", "1.5", "-1", "0x10", "-0", "1_1", " 1", "1 "):
```

Comment:
Can you add a test with Unicode digits here?

Reply:
Done.